buildpack/telemetry/newrelic.py

```diff
@@ -44,7 +49,51 @@ def update_config(m2ee, app_name):
 
 
 def get_new_relic_license_key():
+    """Get the New Relic's license key."""
+    # Old om-prem integration
     vcap_services = util.get_vcap_services_data()
     if vcap_services and "newrelic" in vcap_services:
         return vcap_services["newrelic"][0]["credentials"]["licenseKey"]
-    return None
+
+    # New integration
+    return os.environ.get("NEW_RELIC_LICENSE_KEY", None)
+
+
+def is_enabled() -> bool:
+    """
+    The function checks if all environment variables required for
+    New Relic connection are set up. New integration only.
+    The old om-prem integration does not care about this.
+    """
+    return all(map(os.getenv, REQUIRED_NEW_RELIC_ENV_VARS))
+
+
+def get_metrics_config() -> Dict:
+    """Configs to be used by telegraf. New integration only."""
+    return {
+        "api_key": os.getenv("NEW_RELIC_LICENSE_KEY"),
+        "metrics_base_url": os.getenv("NEW_RELIC_METRICS_API"),
+    }
+
+
+def get_metrics_tags(app_name) -> Dict:
+    """Tags to be used by telegraf. New integration only."""
+    return {
+        "app": util.get_app_from_domain(),
+        "instance_index": int(os.getenv("CF_INSTANCE_INDEX", "0")),
+        "appName": app_name
+    }
```

Comment:
> Are we passing the CP environment_id as metadata in other integrations? I think we don't need to. The name of the application should suffice
Will confirm that.

> Also, let's adopt a single naming style for the tags. Use underscores.
The metrics collected from New Relic's agent have the `appName` column, that's why I used it that way (and it's the environment_id)

> So just app_name and instance_index should be good.
Currently, `app` is the name taken from the domain and `appName` is the environment_id.
- AppDynamics: has at least the environment_id (don't know the field name now)
- Datadog: `application_name` has the environment_id, `service` has the name taken from the domain (for logs at least)
- Splunk: has `application_name` holding the name from the domain, and `environment_id`, with the id... (only logs as well)
- Dynatrace: not sure at the moment

Reply:
> Also, let's adopt a single naming style for the tags. Use underscores.
I tested it and New Relic considers app_name and appName to be different.
Since the New Relic metrics come with appName, I think it's better to keep the custom metrics the same way, so the queries will follow the same pattern.

Reply:
Discussed offline and decided to add only the underscore tags as metadata for logs and custom metrics integration.
Also, we'll be adding the same set of tags to the NR Java agent.